### DISCUSSION TITLE
Configuring cloudformation test-type endpoint or stack parameters?

### DISCUSSION THREAD

**Discussion by msummers-nr** [aws/aws-cli#7167] [Q&A] [answered]
Is there a way to configure the endpoint `test-type` invokes or pass in runtime overrides?

I either need to invoke the `TestEntrypoint` so my mocks are used or pass in the security information that the resource expects in the stack (or account?). What I don't want to do is upload my security info that is in the `inputs` directory while I'm testing locally.

Thanks.

**Comment by aBurmeseDev** [comment]
Hi @msummers-nr, thanks for reaching out. Would you be able to elaborate more on your use case as that would help me understand and provide you with proper guidance? 

I'd also like to suggest reviewing these docs for testing resource types and steps to develop a resource type: 
- [Walkthrough: Develop a resource type](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-walkthrough.html)
- [Testing resource types using contract tests](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test.html)

**Comment by msummers-nr** [comment]
Hi @aBurmeseDev 

The use case is how to run `test-type` without leaking api keys in the `input` files.

With local testing I can specify the entry point and setup mocks before the contract tests run. I also have a set of local `input` files with live keys for end-to-end testing but those never leave my local machine.

What I ended-up doing is using a special api key value to signal using the mocks, this is the api key that goes into the public `input` files.

If `test-type` had the ability to set the entry point and/or envvars that would be a big help.

**Comment by aBurmeseDev** [comment, marked as answer]
Thanks for providing more info, @msummers-nr. Unfortunately, this happens to be an issue with CloudFormation rather than the AWS CLI. 

I suggest posting in [AWS CloudFormation GitHub repo](https://github.com/aws-cloudformation) for help or consider asking in [re:Post](https://repost.aws/) as well.
